### ISSUE TITLE
[FedSharing] Unable to edit/reshare a file shared by link

### ISSUE THREAD

**Issue by mcastroSG** [owncloud/core#25221]
### Steps to reproduce
1. As user A connected to server A share a txt file by link
2. Open link and "add to your owncloud" setting serverB url 
3. As user B connected to server B use browser to edit or reshare the file

### Expected behaviour

Able to reshare or edit this file

### Actual behaviour

Unable to reshare or edit this file

### Server configuration

**Operating system**:
Ubuntu 14.04.1 LTS
**Web server:**
Apache2
**Database:**
Server version: 5.5.46-0ubuntu0.14.04.2 (Ubuntu)
**PHP version:**
PHP 5.5.9-1ubuntu4.14 (cli) (built: Oct 28 2015 01:34:46
**ownCloud version:** (see ownCloud admin page)
ownCloud Enterprise Edition 9.1.0 beta 1 (testing)

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:**
Chrome 51
**Operating system:**
OSX El capitán 10.11.5

Related to :

[#24868](https://github.com/owncloud/core/issues/24868)
[#4996](https://github.com/owncloud/client/issues/4996)
[#22122](https://github.com/owncloud/core/issues/22122)

**Comment by PVince81** [comment]
@mcastroSG when creating the link share, you didn't tick "allow editing" so it doesn't give any editing permissions.

As for reshare permissions, these are not set when sharing by link and there was never a way to set these in the web UI. I think we should not change for now, but we might want to reconsider when looking at CRUDS for links shares: https://github.com/owncloud/core/issues/20699

**Comment by PVince81** [comment]
Also see https://github.com/owncloud/core/issues/24868#issuecomment-227776832

**Comment by PVince81** [comment]
Hmm, regarding reshare OC < 9.1 allow the user to reshare regardless of permissions.
It's only since OC 9.1 that the actualy permissions for resharing are respected as it was missing before.

**Comment by PVince81** [comment]
Talked with @rullzer and there are no reshare permissions for link shares.

**Comment by rullzer** [comment]
To give a bit more context. In the current scope this seems to be the safest option. Else you quickly lose track of who has access etc.

If the PR that convertes mounted link shares into actual federated shares is in then we could chose a different path.

It is kind of a weird discussion because by definition link shares can be reshared (since we can't stop somebody from forwarding a link). However this is not real resharing as it is still the same share.

Long story short. I think for now the safest is to have no reshare permissions set on link shares (so just as @PVince81 said in 1 sentence).
